File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hash_map = {}
        for (index, num) in enumerate(nums):
            if num not in hash_map:
                hash_map[num] = index
            else:
                if abs(hash_map[num] - index) <= k:
                    print(hash_map[num])
                    return True
                else:
                    hash_map[num] = index
        return False

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.rs
use std::collections::HashMap;
impl Solution {
    pub fn contains_nearby_duplicate(nums: Vec<i32>, k: i32) -> bool {
        let mut index_map: HashMap<i32, usize> = HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            if let Some(&prev_index) = index_map.get(&num) {
                if i - prev_index <= k as usize {
                    return true;
                }
            }
            index_map.insert(num, i);
        }

        false
    }
}